src/modal-dialog.cpp

```diff
@@ -169,6 +172,7 @@ void Modal_Dialog::show(const Fl_Widget *p) {
 	}
 #endif
 	while (_dialog->shown()) { Fl::wait(); }
+	Fl::grab(prev_grab);

```

Comment:
If there was already an `Fl::grab()` widget, then opening a modal would cause a softlock. This fixes it.

This is probably only possible on Mac (because the Mac sys menubar can be used to open a modal while (e.g.) the Palette dropdown is open. But the bug is theoretically valid on any platform, so this is a good fix regardless.